dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNServiceManagerInterfaceIIDM.h

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright (c) 2020, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+// simulation tool for power systems.
+//
+
+/**
+ * @file DataInterface/PowSyblIIDM/DYNServiceManagerInterfaceIIDM.h
+ *
+ * @brief Service manager IIDM implementation file
+ *
+ */
+#ifndef MODELER_DATAINTERFACE_POWSYBLIIDM_DYNSERVICEMANAGERINTERFACEIIDM_H_
+#define MODELER_DATAINTERFACE_POWSYBLIIDM_DYNSERVICEMANAGERINTERFACEIIDM_H_
+
+#include "DYNGraph.h"
+#include "DYNServiceManagerInterface.h"
+#include "DYNVoltageLevelInterfaceIIDM.h"
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace DYN {
+
+class DataInterfaceIIDM;
+
+/**
+ * @brief Implementation for IIDM dynawo service
+ */
+class ServiceManagerInterfaceIIDM : public ServiceManagerInterface {
+ public:
+  /**
+   * @brief Constructor
+   *
+   * @param dataInterface data interface corresponding to current service manager
+   */
+  explicit ServiceManagerInterfaceIIDM(const DataInterfaceIIDM* const dataInterface);
```

Comment:
Please used shared_ptr

Reply:
Not in this case. Shared_ptr have ownership of the pointer, which we do not want here. Besides, ServiceManagerInterfaceIIDM constructor is called inside DataInterfaceIIDM interface.